src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -112,7 +112,39 @@ private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
                 List<IEdmStructuralProperty> key = new List<IEdmStructuralProperty>();
                 foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
                 {
-                    IEdmStructuralProperty structuralProperty = this.FindProperty(keyProperty.PropertyName) as IEdmStructuralProperty;
+                    IEdmStructuralProperty structuralProperty = null;
+                    if (!string.IsNullOrEmpty(keyProperty.Alias))
+                    {
+                        string[] keyProperties = keyProperty.PropertyName.Split('/');
+                        for (int i = 0; i < keyProperties.Length; i++)
+                        {
+                            if (i == 0)

```

Comment:
Please unroll the loop so that we don't have to do this check for every subsequent iteration

Reply:
Actually, the body of these two `if` statements look the same to me. Why do we need this check here?